src/utils/tools.py

```diff
@@ -248,3 +253,56 @@ def log(self, *args, **kwargs):
     def close(self):
         if self.logfile_output_stream:
             self.logfile_output_stream.close()
+
+
+def initialize_data_loaders(
+    dataset: torch.utils.data.Dataset,
+    data_indices: list[dict[str, list[int]]],
+    batch_size: int=32,
+) -> tuple[DataLoader, DataLoader, DataLoader, torch.utils.data.Subset, torch.utils.data.Subset, torch.utils.data.Subset]:
```

Comment:
My thought of extracting this as a utility function was that this makes the usage of the function more flexible. You could in theory also import it on the clients and use it there e.g. for the creation of auxiliary data loaders in a distillation setting.

Reply:
Sounds fair. Modify the type hints to use `Dataset` and `Subset` directly can be better, since somehow python plugin on vscode is not able to recognize `torch.utils.*` modules.

Reply:
Sounds fair. Modify the type hints to use Dataset and Subset directly can be better, since somehow python plugin on vscode is not able to recognize torch.utils.* modules.